web3/_utils/formatters.py

```diff
@@ -41,12 +55,14 @@ def apply_formatters_to_args(*formatters):
 
 @curry
 @to_list
-def apply_formatter_to_array(formatter, value):
+def apply_formatter_to_array(
+    formatter: Callable[[TValue], TReturn], value: Collection[TValue]
```

Comment:
I think this should be `Sequence` instead of `Collection` since there is an assumption of ordering which a collection doesn't guarantee.